tensorflow/models/image/alexnet/alexnet_benchmark.py

```diff
@@ -198,10 +255,14 @@ def run_benchmark():
     sess.run(init)
 
     # Run the forward benchmark.
-    time_tensorflow_run(sess, pool5, "Forward")
-
-    # Add a simple objective so we can calculate the backward pass.
-    objective = tf.nn.l2_loss(pool5)
+    time_tensorflow_run(sess, logits, "Forward")
+
+    # Add the objective so we can calculate the backward pass.
+    if FLAGS.l2_loss:
+      objective = tf.nn.l2_loss(logits)
+    else:
+      objective = tf.reduce_mean(tf.nn.softmax_cross_entropy_with_logits(
+          logits, labels))
     # Compute the gradient with respect to all the parameters.
     grad = tf.gradients(objective, parameters)
```

Comment:
So, here intead of running with all grad being the target, we can have a "sink" that will not require memcpy. This can be achieved by:
```python
sink = tf.group(*grad)
time_tensorflow_run(sess, sink, "Forward-backward")
```

Note that this matches training time behavior, since we most likely would want everything to stay on the GPU during training, so no memcpy will be needed effectively.